### PR TITLE
chaos: configure http fault percent fraction denominator

### PR DESCRIPTION
**Pull request by djoodle**
### Description
Changing the Chaos experimentation HTTPFaultConfig to support specifying the fraction denominator for FaultPercentage to allow for greater precision. 

### Testing Performed
The changes are unit tested

### CODE REVIEW

**Review by bwitt — APPROVED** [review]

**Comment by bwitt** [comment]
I assume this updates the UI too; what does that look like with this change?

File: api/chaos/serverexperimentation/v1/serverexperimentation.proto [hunk 1]
```diff
@@ -111,9 +111,31 @@ message ClusterPercentage {
...
+    // **Example**: 1/1000000 = 0.0001%.
+    DENOMINATOR_MILLION = 3;
+  }
+
+  // The numerator of a percentage.
+  // (i.e. percentage equal to 50 with denominator set to HUNDRED results in 50/100 = 50%)
+  uint32 percentage = 1 [ (validate.rules).uint32 = {gt : 0, lte : 1000000} ];
+
+  // Specifies the denominator. If the denominator specified is less than the numerator, the final
+  // fractional percentage is capped at 1 (100%). Unspecified denominaotr will be handled as 100.
```
Comment (snowp) [inline api/chaos/serverexperimentation/v1/serverexperimentation.proto:137]: typo in `denominaotr`

File: backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go [hunk 2]
```diff
@@ -171,6 +171,10 @@ func (g ECDSFaultsGenerator) createAbortDelayConfig(httpFaultConfig *serverexper
 	var abort *gcpFilterFault.FaultAbort
 	var delay *gcpFilterCommon.FaultDelay
 
+	pcntage, err := GetHTTPFaultPercentage(httpFaultConfig)
```
Comment (snowp) [inline backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go:174]: nit: `percentage`, doesn't hurt to spell it out :)

**Comment by djoodle** [comment]
> I assume this updates the UI too; what does that look like with this change?

This updates the proto definitions in the frontend - but actually doesn't need any UI changes. If no denominator is provider it assumes HUNDRED - which was previous behaviour.  Typing of numerator hasn't change - just we don't enforce it to be 100 or less - but reading the envoy docs providing a numerator > denominator = 100%

**Review by snowp — APPROVED** [review]
LGTM, few nits. It'd be good to track UI changes so that this can be configured via the UI as well, but not strictly necessary

File: backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go (hunk 2 above)
Comment (djoodle) [reply backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go:174]: Ha you aint seen my keyboard - every character is a pain xD

**Comment by djoodle** [comment]
Fix nits

**Pull request merged**